app/models/mixins/archived_purging_mixin.rb

```diff
@@ -0,0 +1,41 @@
+module ArchivedPurgingMixin
+  extend ActiveSupport::Concern
+  include PurgingMixin
+
+  module ClassMethods
+    def purge_date
+      ::Settings.archived_entities.history.keep_archived_entities.to_i_with_method.seconds.ago.utc
+    end
+
+    def purge_window_size
+      ::Settings.archived_entities.history.purge_window_size
+    end
+
+    def purge_queue(ts)
+      MiqQueue.put_unless_exists(
+        :class_name  => name,
+        :method_name => "purge",
+        :queue_name  => "ems",
+        :args        => [ts]
+      )
+    end
+
+    alias_method :purge_timer, :purge_queue
+
+    def purge_timer
```

Comment:
Either use `alias_method` OR define `purge_timer`

if you go with `alias_method`, add a `ts ||= purge_date`